pynxtools/dataconverter/readers/stm/reader.py

```diff
@@ -0,0 +1,228 @@
+"""
+    A short description on SPM reader.
+"""
+
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from typing import Any, Dict, Tuple, Union
+from collections.abc import Callable
+import json
+import yaml
+
+from pynxtools.dataconverter.readers.base.reader import BaseReader
+from pynxtools.dataconverter.template import Template
+from pynxtools.dataconverter.readers.stm.sts_file_parser import from_dat_file_into_template
+from pynxtools.dataconverter.readers.stm.stm_file_parser import STM_Nanonis
+from pynxtools.dataconverter.readers.utils import flatten_and_replace, FlattenSettings
+
+
+CONVERT_DICT = {
+    'Instrument': 'INSTRUMENT[instrument]',
+    'Software': 'SOFTWARE[software]',
+    'Hardware': 'Hardware[hardware]',
+    'Analyser': 'ELECTRONANALYSER[electronanalyser]',
+    'Beam': 'BEAM[beam]',
+    'unit': '@units',
+    'version': '@version',
+    'Sample': 'SAMPLE[sample]',
+    'User': 'USER[user]',
+    'Data': 'DATA[data]',
+    'Source': 'SOURCE[source]',
+    'Environment': 'ENVIRONMENT[environment]',
+    'Sample_bias': 'SAMPLE_BIAS[sample_bias]'
+}
+
+REPLACE_NESTED: Dict[str, str] = {}
+
+
+# pylint: disable=too-few-public-methods
+class StmNanonisGeneric5e:
+    """Class to handle 'stm' experiment of software version 'Generic 5e' from 'nanonis'
```

Comment:
okay now I understand that for now all your stm is specifically for nanonis software, generic 5e